### PR TITLE
Feature/websockets using websocketpp

### LINKED ISSUE #20: Support WebSocket transport
via: https://github.com/zaphoyd/websocketpp

### PR DESCRIPTION
**Pull request by DZabavchik**
Uses WebSocket++ (https://github.com/zaphoyd/websocketpp 0.6.0 or 0.7.0).
Tested on OS X, and Windows
Fixes #20

-- Need to test TLS (wss://)

### CODE REVIEW

**Comment by jpetso** [comment]
This branch seems to contain a lot of changes that were already merged into autobahn-cpp. If it were my repo, I'd ask you squash & rebase on top of current master so all of the commits unrelated to the websocket feature disappear from the merged branch.

**Comment by DZabavchik** [comment]
Thanks @jpetso. Bad habits of keeping history die hard.

File: autobahn/wamp_websocket_transport.hpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) Tavendo GmbH
+// Copyright (c) NinjaRMM, LLC
```
Comment (oberstet) [inline autobahn/wamp_websocket_transport.hpp]: Please consider https://github.com/crossbario/autobahn-cpp/issues/107

Comment (DZabavchik) [inline autobahn/wamp_websocket_transport.hpp]: Ok. np

File: examples/SConscript [hunk 2]
```diff
@@ -44,7 +45,7 @@ prgs = []
 
 for e, extralibs in examples:
     prgs.append(env.Program(e, LIBS=[
-        'boost_thread',
+        'boost_thread-mt',
```
Comment (DZabavchik) [inline examples/SConscript]: Is this ok?  Couldn't get it to work without `-mt`

Comment (oberstet) [inline examples/SConscript]: I have no clue;) But http://stackoverflow.com/questions/3031768/boost-thread-linking-boost-thread-vs-boost-thread-mt seems to suggest it's a symlink anyway .. let's try it.

**Comment by oberstet** [comment]
@DZabavchik coool! sorry for late response ... I was busy, then started to review, got interrupted again. Anyway: I like it! Thanks in particular for caring about both cmake and scons, and for adding an example too! IOW: I'd merge -- if you could change the copyright notice - pls see the discussion I linked - we want to give credits of course, but it needs to be sustainable. Maybe a CHANGELOG file would be a way to give visibility to contributors by name. I'd be ready for that.

**Comment by DZabavchik** [comment]
I changed the copyright, probably while you were typing last comment since my commit appears above your comment :)

Whatever you do with CHANGELOG or not, I have no pref. Autobahn-cpp now does what I need it to (prefix regs, Windows, and websocket transport).  I'm happy about it and grateful to other contributors.  I thought it wouldn't hurt to do it right (scons and at least one example). 

I'd like to see you weight in on protocol issue #182 (https://github.com/wamp-proto/wamp-proto/issues/182) as original author of the feature (intent, use cases).  I staged the suggested changes and got same performance with prefix registrations as with exact matches. Thanks.

**Comment by oberstet** [comment]
@DZabavchik Awesome!! Thanks a lot. 

Rgd changelog and contributor credits: https://github.com/crossbario/autobahn-cpp/issues/108

**Pull request merged**